infrastructure/templates/public-api/application/shared/recoveryVault.bicep

```diff
@@ -0,0 +1,55 @@
+import { ResourceNames } from '../../types.bicep'
+
+@description('Common resource naming variables.')
+param resourceNames ResourceNames
+
+@description('The location to create resources in.')
+param location string
+
+@description('Whether manual deletion of backups is allowed.')
+param immutable bool
+
+@description('Tags for the resources')
+param tagValues object
+
+module recoveryVaultModule '../../components/recoveryVault.bicep' = {
+  name: 'recoveryVaultDeploy'
+  params: {
+    vaultName: resourceNames.sharedResources.recoveryVault
+    location: location
+    redundancy: 'GeoRedundant'
+    immutable: immutable
+    softDelete: {
+      retentionDays: 30
+      state: 'AlwaysON'
+    }
+    crossSubscriptionRestoreEnabled: false
+    crossRegionRestoreEnabled: true
+    alertsEnabled: true
+    tagValues: tagValues
+  }
+}
+
+module fileShareBackupPolicyModule '../../components/recoveryVaultFileShareBackupPolicy.bicep' = {
+  name: 'fileShareBackupPolicyDeploy'
+  params: {
+    policyName: resourceNames.sharedResources.recoveryVaultFileShareBackupPolicy
+    vaultName: resourceNames.sharedResources.recoveryVault
+    location: location
+    dailyBackupTimeUtc: '00:00'
+    daysToRetainDailyBackups: 30
+    weeklySnapshotRetention: {
+      targetBackupDay: 'Monday'
+      weeksToRetain: 8
+    }
+    monthlySnapshotRetention: {
+      targetBackupDay: 'Monday'
+      targetBackupWeek: 'First'
+      monthsToRetain: 12
+    }
```

Comment:
I didn't see anything in the config about incremental vs. full backups. Do you know if incremental backups are an option and did you consider them?

I was also wondering if you thought about how the size of the file share might grow and looked at the cost of storage? I think storage is cheap anyway, so hopefully should not be an issue! 🙂